### PR TITLE
[docker] Add a number of types to docker.models.containers.Container

### PR DESCRIPTION
**Pull request by adamtheturtle**

### CODE REVIEW

File: stubs/docker/docker/api/container.pyi [hunk 1]
```diff
@@ -154,10 +154,10 @@ class ContainerApiMixin:
     def rename(self, container: _Container, name: str) -> None: ...
     def resize(self, container: _Container, height: int, width: int) -> None: ...
     def restart(self, container: _Container, timeout: int = 10) -> None: ...
-    def start(self, container: _Container, *args, **kwargs) -> None: ...
+    def start(self, container: _Container) -> None: ...
```
Comment (adamtheturtle) [inline stubs/docker/docker/api/container.pyi:157]: An exception is raised if any args or kwargs are given: https://github.com/docker/docker-py/blob/a3652028b1ead708bd9191efb286f909ba6c2a49/docker/api/container.py#L1128-L1133

File: stubs/docker/docker/models/containers.pyi [hunk 2]
```diff
@@ -71,18 +71,32 @@ class Container(Model):
...
+    def pause(self) -> None: ...
     def put_archive(self, path: str, data) -> bool: ...
-    def remove(self, **kwargs) -> None: ...
+    def remove(self, *, v: bool = False, link: bool = False, force: bool = False) -> None: ...
     def rename(self, name: str): ...
     def resize(self, height: int, width: int): ...
-    def restart(self, **kwargs): ...
-    def start(self, **kwargs) -> None: ...
+    def restart(self, *, timeout: float | None = 10): ...
+    def start(self) -> None: ...
```
Comment (adamtheturtle) [inline stubs/docker/docker/models/containers.pyi:80]: See https://github.com/python/typeshed/pull/12077/files#r1623166228 for why I chose to remove `**kwargs`

File: stubs/docker/docker/api/container.pyi [hunk 3]
```diff
@@ -157,7 +157,7 @@ class ContainerApiMixin:
     def start(self, container: _Container, *args, **kwargs) -> None: ...
```
Comment (adamtheturtle) [inline stubs/docker/docker/api/container.pyi:157]: At https://github.com/docker/docker-py/blob/a3652028b1ead708bd9191efb286f909ba6c2a49/docker/api/container.py#L1128-L1133 we see:

```python
        if args or kwargs:
            raise errors.DeprecatedMethod(
                'Providing configuration in the start() method is no longer '
                'supported. Use the host_config param in create_container '
                'instead.'
            )
```

Tips are welcome on how to change this signature and not break CI.

Comment (srittau) [reply stubs/docker/docker/api/container.pyi:157]: You can just remove the arguments and then add the function to `@tests/stubtest_allowlist.txt` (with a comment) to silence stubtest.

Comment (adamtheturtle) [reply stubs/docker/docker/api/container.pyi]: Thank you @srittau , I have done this.

**Comment by adamtheturtle** [comment]
@srittau Please could you take another look at this when you have time? I'm also happy to split it up if that makes a review easier.

**Review by srittau — APPROVED** [review]
Thanks!

**Pull request merged**